sequence-model-generator/modules/sequence-model-generator-core/src/main/java/io/ballerina/sequencemodelgenerator/core/CommonUtil.java

```diff
@@ -0,0 +1,130 @@
+/*
+ *  Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com)
+ *
+ *  WSO2 LLC. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package io.ballerina.sequencemodelgenerator.core;
+
+import io.ballerina.compiler.api.symbols.Symbol;
+import io.ballerina.compiler.api.symbols.TypeDescTypeSymbol;
+import io.ballerina.compiler.api.symbols.TypeReferenceTypeSymbol;
+import io.ballerina.compiler.api.symbols.TypeSymbol;
+import io.ballerina.compiler.api.symbols.UnionTypeSymbol;
+import io.ballerina.compiler.syntax.tree.ModulePartNode;
+import io.ballerina.compiler.syntax.tree.NonTerminalNode;
+import io.ballerina.compiler.syntax.tree.SyntaxTree;
+import io.ballerina.projects.DocumentId;
+import io.ballerina.projects.Project;
+import io.ballerina.tools.text.TextRange;
+
+import java.nio.file.Path;
+import java.util.Optional;
+
+/**
+ * Represents the common utility functions for the sequence diagram.
+ *
+ * @since 2201.9.0
+ */
+public class CommonUtil {
+
+    private CommonUtil() {
+    }
```

Comment:
```suggestion
```
No need for a private constructor because all the methods are static methods.

Reply:
This is a common mechanism used for utility classes with static methods to ensure that an instance cannot be created from them.